### PR TITLE
feat(server/build-detector): inform user of missed build step

### PR DESCRIPTION
**Pull request by TasoOneAsia**
**Pull Request Description**

Explicitly convey to the user that using the release is a required step, literally giving them a direct DL link to the latest GH release.

**Pull Request Checklist**:
* [x] Have you followed the guidelines in our Contributing document and Code of Conduct?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/project-error/npwd/pulls) for the same update/change?
* [x] Have you built and tested NPWD in-game after the relevant change?

### CODE REVIEW

File: build-detector.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+on('onResourceStart', async (resName) => {
+  if (resName !== currentResourceName && process.env.NODE_ENV !== 'development') return;
+
+  const currentResourcePath = GetResourcePath(currentResourceName);
+  const targetBuildPath = path.join(currentResourcePath, 'resources', 'dist');
+
+  try {
+    await fs.promises.access(targetBuildPath);
+  } catch {
```
Comment (RockySouthpaw) [inline build-detector.js:20]: Should we also toss a confirmation message saying it contains built files or something letting them know they can read. Easier for debugging.

File: build-detector.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    console.log(
+      '^1==============================================\n\n' +
+        '!!! NPWD WAS UNABLE TO LOCATE BUILT FILES AND WAS UNABLE TO START !!!\n\n' +
+        'This error is most likely caused by downloading the source code instead of the release\n\n' +
```
[the remark this answers is not in the document]
Comment (TasoOneAsia) [reply build-detector.js:24]: Can also arise due to error in build process.

File: build-detector.js (hunk 1 above)
Comment (TasoOneAsia) [reply build-detector.js:20]: Services are logged under the "info" level when they start.

File: build-detector.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        'Find the latest NPWD release here: \n\n' +
+        'https://github.com/project-error/npwd/releases/latest/download/npwd.zip\n\n' +
```
[the remark this answers is not in the document]
Comment (TasoOneAsia) [reply build-detector.js:26]: Opening a browser tab during startup process might tick off server owners a good amount, could be something to consider but should be doing this in a way that does not mess with the host machine any.

[the remark this answers is not in the document]
Comment (RockySouthpaw) [reply build-detector.js:26]: If only there was clipboard api

File: build-detector.js (hunk 1 above)
Comment (RockySouthpaw) [reply build-detector.js:20]: image

Yea but they're quite obnoxious and i've never recalled a time where these actually said they didn't start lol.

Comment (RockySouthpaw) [reply build-detector.js:20]: Meh fuck it

**Comment by RockySouthpaw** [comment]
There’s also a potential issue with this in terms of being misleading or server owners not seeing this print, though it’s not really a fault on the resource. 

The error message states NPWD failed to start however, it’s actually started. So any resource, like esx-npwd or custom frameworks, may encounter fatal issues if they’re relying on `if GetResourceState('npwd') == 'started' then` to let the resource know the resource was started as any sort of error with whatever pertains to the above check will result in the subsequent code not being executed.

This will also result in subsequent errors due to a lack of building like “no export newPlayer” which would also give us a bit of a harder time with support as people may be asking why framework integration isn’t working.

Ideally, we would stop the resource since it wasn’t built correctly to prevent any errors or misbehaving of reliant resources. 

If you’re thinking “how will a server owner not see a multi line print?” you haven’t seen some of the console logs these server owners have:

**Comment by RockySouthpaw** [comment]
There’s also some cases of extremely obnoxious resources doing multi line prints, like essentialmode for instance, that just spells their names or whatever branding. We should also account for those people when implementing this so it may be worth it to run a thread if we can’t stop the resource altering the server owner of their issue.

**Comment by TasoOneAsia** [comment]
This is as far as we can go in terms of handling for building. Anything more and it becomes tediously interdependent and will devolve into spaghetti over time.

**Comment by RockySouthpaw** [comment]
Closed in favor of https://github.com/project-error/npwd/pull/559 due to me breaking this kek

**Pull request closed without merging**